### PR TITLE
Address a number of TODO(3.0) comments in the code

### PR DESCRIPTION
**Pull request by paulidale**
- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/params_from_text.c [hunk 1]
```diff
@@ -75,8 +75,8 @@ static int prepare_from_text(const OSSL_PARAM *paramdefs, const char *key,
         *buf_n = (buf_bits + 7) / 8;
 
         /*
-         * TODO(v3.0) is this the right way to do this?  This code expects
-         * a zero data size to simply mean "arbitrary size".
+         * A zero data size to simply mean "arbitrary size", so only do the
```
Comment (slontis) [inline crypto/params_from_text.c:78]: Comment is not right now..

Comment (paulidale) [reply crypto/params_from_text.c]: Fixed.

**Review by slontis — APPROVED** [review]
Apart from a NIT this looks fine.
Second opinion probably needed..

**Comment by t8m** [comment]
@paulidale This PR overlaps with a PR I did and I also do not think it is correct in some aspects. Please take out the provider_core.c change.

**Comment by t8m** [comment]
Or you can just rebase now.

**Comment by paulidale** [comment]
I've rebased and removed the provider_core.c patch.

**Comment by paulidale** [comment]
Merged to master, thanks for the comments and reviews.

**Pull request closed without merging**